### ISSUE TITLE
Proposal: Consistent package naming convention

### ISSUE THREAD

**Issue by loewenm** [dmlc/xgboost#10048]
Greetings dmlc

The naming convention of XGBoost is inconsistent with how nearly all other python packages are handled.

**Situation:**

When installing XGBoost, we refer to the package as "py-xgboost" given the cli instruction as follows:

`conda install -c conda-forge py-xgboost`

This installs the latest package:

![Screenshot 2024-02-15 085306](https://github.com/dmlc/xgboost/assets/19495222/c72d2af3-ee3c-459f-863b-a275c96e4129)
(i.e. py-xgboost v.1.7.3)

**Problem**

How we reference this package changes from here on out for some reason...

In order to print the current version of py-xgboost, we need to call xgboost (i.e. not py-xgboost).

![Screenshot 2024-02-15 085633](https://github.com/dmlc/xgboost/assets/19495222/93c66a2e-7e56-4349-ad14-02ad160caf10)

In fact, we _import_ xgboost in our jupyter notebooks or python files by calling "import xgboost".

**Solution**

Have a naming convention that is consistent between the package name and all future references.

Pick either "py-xgboost" or "xgboost"

Thank you for your consideration.

**Comment by hcho3** [comment]
Hello,

I acknowledge the merit of consistent naming. Unfortunately, there are reasons for the discrepancy in the package naming for XGBoost. Let me explain:

* Conda-forge is a **third-party** distribution channel, with its own naming conventions. The channel we (XGBoost maintainers) directly control is PyPI: https://pypi.org/project/xgboost/. Here, we use the "correct" name (`xgboost`).
* XGBoost is not purely a Python package; it consists of multiple parts, including the C++ code and the R language binding.
* When Conda-forge distributes multiple components of a software, it uses prefixes to distinguish the components. Conda-forge currently distributes three components of XGBoost: `libxgboost` (C++ core), `r-xgboost` (R package), and `py-xgboost` (Python package). You can see a similar policy applied to other packages, e.g. `python-graphviz` for the Python interface of Graphviz.

In short, the naming of `py-xgboost` is due to the policy of the Conda-forge distribution channel and we (XGBoost maintainers) cannot change that.

**Comment by loewenm** [comment]
Thank you @hcho3. I knew there had to be something to do with the different languages (C++ and R). Makes sense.
